File: taskhub_app/lib/models/subtarea.dart
/// Un ítem de checklist dentro de una tarea.
class SubTarea {
  final int id;
  final String titulo;
  final bool completada;

  SubTarea({
    required this.id,
    required this.titulo,
    required this.completada,
  });

  factory SubTarea.fromJson(Map<String, dynamic> json) => SubTarea(
        id: json['id'] as int,
        titulo: json['titulo'] as String,
        completada: json['completada'] as bool? ?? false,
      );
}

File: TaskHub.API/Models/SubTarea.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace TaskHub.API.Models;

/// <summary>
/// Un ítem de checklist dentro de una tarea.
/// </summary>
public class SubTarea
{
    public int Id { get; set; }

    [Required]
    [MinLength(1)]
    public string Titulo { get; set; } = string.Empty;

    public bool Completada { get; set; }

    public int TaskItemId { get; set; }

    [JsonIgnore]
    public TaskItem? TaskItem { get; set; }
}
